### ISSUE TITLE
[Core feature] Different resources for Ray head pod and worker pods

### ISSUE THREAD

**Issue by aybidi** [flyteorg/flyte#3885]
### Motivation: Why do you think this is important?

We are running ML workflows at my organization that often require spinning up a Ray cluster where the head nodes and worker nodes usually have different resources configuration. Using the Ray plugin, I could use the `limits` arg in the `@task` decorator to set resources for the task.

```python
ray_config = RayJobConfig(
    head_node_config=HeadNodeConfig(...),
    worker_node_config=[WorkerNodeConfig(...),],
    runtime_env={...},
)
@task(task_config=ray_config, limits=Resources(mem="12Gi", cpu="4"))
def ray_task(...):
    ...
```

However, this way of setting resources sets the same resources for both the head pod and worker pods CR created by the backend Ray plugin.

### Goal: What should the final outcome look like, ideally?

Users should be able to control the cluster specs at a more granular level i.e., specifying different specs for the head node and the worker nodes.

### Describe alternatives you've considered 

n/a

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by pingsutw** [comment]
Yes, we can support it. We need to update flyteidl and flyteplugins. You can check out these PRs that enables us to set different resource in the kubeflow task.
- https://github.com/flyteorg/flyteplugins/commit/9a2bbbaf2f3ac9e38222ba7755ac4bda1ac2fdb7
- https://github.com/flyteorg/flyteidl/commit/5a3a44f13415bf5ed8ac57cc5814b7277a85825f

**Comment by Sovietaced** [comment]
This landed in Flyte v1.14.0: https://github.com/flyteorg/flyte/releases/tag/v1.14.0
